osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/ClearAlarmRegisterCommandExecutorTest.java

```diff
@@ -194,18 +200,108 @@ void successRegister1AndResultAlarmRegister2() throws ProtocolAdapterException,
     assertThat(accessResultCode).isEqualTo(AccessResultCode.TEMPORARY_FAILURE);
   }
 
+  @Test
+  void resultAlarmRegister2() throws ProtocolAdapterException, IOException {
+    when(this.dlmsConnection.set(this.setParameterArgumentCaptor.capture()))
+        .thenReturn(AccessResultCode.SUCCESS)
+        .thenReturn(AccessResultCode.SUCCESS);
+
+    final DlmsDevice dlmsDevice = new DlmsDevice("SMR 5.2 device");
+    this.setupAlarmRegister2(dlmsDevice);
+    final AccessResultCode accessResultCode =
+        this.executor.execute(this.connectionManager, dlmsDevice, this.dto, this.messageMetadata);
+    assertThat(accessResultCode).isEqualTo(AccessResultCode.SUCCESS);
+  }
+
+  @Test
+  void connectionProblemAlarmRegister3() throws ProtocolAdapterException, IOException {
+    when(this.dlmsConnection.set(any(SetParameter.class)))
+        .thenReturn(AccessResultCode.SUCCESS)
+        .thenReturn(AccessResultCode.SUCCESS)
+        .thenThrow(new IOException());
+
+    final DlmsDevice dlmsDevice = new DlmsDevice("SMR 5.5 device");
+    this.setupAlarmRegister3(dlmsDevice);
+    final Throwable actual =
+        catchThrowable(
+            () ->
+                this.executor.execute(
+                    this.connectionManager, dlmsDevice, this.dto, this.messageMetadata));
+    assertThat(actual).isInstanceOf(ConnectionException.class);
+  }
+
+  @Test
+  void nullResultAlarmRegister3() throws ProtocolAdapterException, IOException {
+    when(this.dlmsConnection.set(this.setParameterArgumentCaptor.capture()))
+        .thenReturn(AccessResultCode.SUCCESS)
+        .thenReturn(AccessResultCode.SUCCESS)
+        .thenReturn(null);
+
+    final DlmsDevice dlmsDevice = new DlmsDevice("SMR 5.5 device");
+    this.setupAlarmRegister3(dlmsDevice);
+    final Throwable actual =
+        catchThrowable(
+            () ->
+                this.executor.execute(
+                    this.connectionManager, dlmsDevice, this.dto, this.messageMetadata));
+    assertThat(actual).isInstanceOf(ProtocolAdapterException.class);
+  }
+
+  @Test
+  void failureRegister2AndResultAlarmRegister3() throws ProtocolAdapterException, IOException {
```

Comment:
Is this test working ok? I would expect it to fail, see my comment in the command executor.